### PR TITLE
✨ Add a client listener to TrackingHttpClient 

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

The client listener allows users to provide attributes to a RUM Resource by inspecting the `HttpClientRequest` and `HttpClientResponse`.

One issue with this approach is that HttpClientRequest and HttpClientResponse do not have much information related to the body of the request and response. Unfortunately, without listening into the  stream and keeping a copy of all written data, we can't easily return this information to the user. 

refs: RUMM-3297, #397

### Review checklist

- [x] This pull request has appropriate unit and / or integration tests 
- [x] This pull request references a Github or JIRA issue

### CI Configuration (optional)
- [x] Run unit tests
- [x] Run integration tests

### CODE REVIEW

File: packages/datadog_tracking_http_client/lib/datadog_tracking_http_client.dart [hunk 1]
```diff
@@ -28,20 +30,24 @@ extension TrackingExtension on DdSdkConfiguration {
   /// resources traced in this way is determined by
   /// [RumConfiguration.tracingSamplingRate].
   ///
+  /// You can add attributes to RUM Resources by providing an
+  /// [clientListener]. See [DatadogTrackingHttpClientListener] for more info.
```
Comment (0xnm) [inline packages/datadog_tracking_http_client/lib/datadog_tracking_http_client.dart:34]:
```suggestion
  /// You can add attributes to RUM Resources by providing a
  /// [clientListener]. See [DatadogTrackingHttpClientListener] for more info.
```

File: packages/datadog_tracking_http_client/lib/src/tracking_http_client_plugin.dart [hunk 2]
```diff
@@ -9,8 +9,45 @@ import 'package:datadog_flutter_plugin/datadog_internal.dart';
 
 import '../datadog_tracking_http_client.dart';
 
+/// An interface for providing attributes to Datadog RUM resources by listening
+/// to HttpClient requests and responses
```
Comment (0xnm) [inline packages/datadog_tracking_http_client/lib/src/tracking_http_client_plugin.dart:13]:
```suggestion
/// to HttpClient requests and responses.
```

File: packages/datadog_tracking_http_client/lib/src/tracking_http_client_plugin.dart [hunk 3]
```diff
@@ -9,8 +9,45 @@ import 'package:datadog_flutter_plugin/datadog_internal.dart';
...
+///
+/// DatadogTrackingHttpClient allows you to receive a callback when an
+/// HttpClientRequest starts and when an HttpClientResponse finishes. They
+/// provide a resource key and a mutable Map<String, Object?> of attributes that
+/// you can modify to add attributes the the resulting Datadog RUM resource.
```
Comment (0xnm) [inline packages/datadog_tracking_http_client/lib/src/tracking_http_client_plugin.dart:18]: probably you've meant something like this:

```suggestion
/// you can modify to add attributes to the resulting Datadog RUM resource.
```

**Review by 0xnm — APPROVED** [review]
lgtm, added few spelling suggestions.

**Review by xgouchet — APPROVED** [review]

**Pull request merged**